### ISSUE TITLE
Is Future support planned?

### ISSUE THREAD

**Issue by vi** [goandylok/comp-rs#3]
> Pure-macro Do notation and List-comprehension for Option, Result and Iterator.

Will Future and Stream also appear on the list as a logical continutaion?

**Comment by andylokandy** [comment]
Yes, I planed to support `Future` and `Stream` later. But I am not sure how many people want this feature.
